src/pruna/evaluation/metrics/aesthetic_laion.py

```diff
@@ -0,0 +1,196 @@
+from huggingface_hub.utils import EntryNotFoundError, RepositoryNotFoundError
+from transformers import CLIPImageProcessor, CLIPVisionModelWithProjection
+
+from pruna.engine.utils import set_to_best_available_device
+from pruna.evaluation.metrics.metric_stateful import StatefulMetric
+from pruna.evaluation.metrics.registry import MetricRegistry
+from pruna.evaluation.metrics.result import MetricResult
+from pruna.evaluation.metrics.utils import metric_data_processor
+from pruna.logging.logger import pruna_logger
+
+METRIC_AESTHETIC_LAION = "aesthetic_laion"
+
+
+@MetricRegistry.register(METRIC_AESTHETIC_LAION)
+class AestheticLAION(StatefulMetric):
+    """
+    Predicts an image aesthetic quality score using LAION-Aesthetics_Predictor V1.
+
+    This metric computes CLIP image embeddings and feeds them into a pretrained
+    linear head released by LAION (matched to the chosen CLIP variant). The model
+    returns a scalar score per image (on a ~1–10 scale). The metric
+    aggregates scores across updates by reporting their mean. Higher is better.
+
+    Parameters
+    ----------
+    *args : Any
+        Additional arguments to pass to the StatefulMetric constructor.
+    device : str | torch.device | None, optional
+        The device to be used, e.g., 'cuda' or 'cpu'. Default is None.
+        If None, the best available device will be used.
+    model_name_or_path : Literal[
+            "openai/clip-vit-large-patch14", "openai/clip-vit-base-patch32", "openai/clip-vit-base-patch16"]
+        The variant of a CLIP model to be used.
+    **kwargs : Any
+        Additional keyword arguments to pass to the StatefulMetric constructor.
+
+    References
+    ----------
+    LAION-Aesthetics_Predictor V1: https://github.com/LAION-AI/aesthetic-predictor
+    """
+
+    total: torch.Tensor
+    count: torch.Tensor
+    call_type: str = "y"
+    higher_is_better: bool = True
+    metric_name: str = METRIC_AESTHETIC_LAION
+
+    model_name_to_aesthetic_head_name: Dict[str, str] = {
+        "openai/clip-vit-large-patch14": "vit_l_14",
+        "openai/clip-vit-base-patch32": "vit_b_32",
+        "openai/clip-vit-base-patch16": "vit_b_16",
+    }
+
+    def __init__(
+        self,
+        *args,
+        device: str | torch.device | None = None,
+        model_name_or_path: Literal[
+            "openai/clip-vit-large-patch14", "openai/clip-vit-base-patch32", "openai/clip-vit-base-patch16"
+        ] = "openai/clip-vit-large-patch14",
+        **kwargs,
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        if model_name_or_path not in self.model_name_to_aesthetic_head_name:
+            pruna_logger.error(f"Model {model_name_or_path} does not exist.")
+            raise ValueError(f"Model {model_name_or_path} does not exist.")
+
+        self.device = set_to_best_available_device(device)
+        try:
+            model_info(model_name_or_path)
+        except (EntryNotFoundError, RepositoryNotFoundError):
```

Comment:
why are we adding this if it should never happen/ also we can just rely on the from pretrained methods lateron, right?